trie-db/src/nibble/mod.rs

```diff
@@ -152,7 +152,7 @@ pub mod nibble_ops {
 #[cfg_attr(feature = "std", derive(Debug))]
 #[derive(Clone, PartialEq, Eq)]
 pub struct NibbleVec {
-	inner: ElasticArray36<u8>,
+	inner: SmallVec<[u8; 36]>,
```

Comment:
This type is used a lot and the hardcoded 36 kind of bothers me. Could you typedef with `pub type ByteVec = SmallVec<[u8; 36]>`.

Naming-wise, I feel like the name `nibble::ByteVec` is probably fine or maybe `BackingVec` or something like that.

Reply:
Yeah, I was on the fence on that, agree it's noisy. 
I did something similar to what you suggest [here](https://github.com/paritytech/parity-common/pull/282/files#diff-7fe73f2989f300cb5d5dc23a9a54a212R31). Reason I had doubts is that the type sort of seeps through and ends up everywhere, so maybe it's better to be explicit about it?
Happy to change this though.

Reply:
Yeah, I'd prefer the typedef. And the `DBKey` type makes sense to me as well.